.github/workflows/ci.yml

```diff
@@ -7,7 +7,87 @@ on:
     tags:
       - '[0-9]+.[0-9]+.[0-9]+'
 
+permissions:
+  contents: read
+
+env:
+  LINE_LENGTH: 99  # TODO: would we like to increase this to 120?
+
 jobs:
+  docker-lint-hadolint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hadolint/hadolint-action@v3.1.0
+        with:
+          ignore: DL3008
+          recursive: true
```

Comment:
Seperate jobs which each do a checkout, etc, versus one job where all the different linting is done in separate steps.

Or see if we can just run the pre-commit hooks here instead using:
`pre-commit run --config './.pre-commit-config.yaml' --all-files`

Reply:
We can try this out, applied it.